shared/code/tce_functions_tcecode.php

```diff
@@ -237,14 +237,14 @@ function F_latex_callback($matches)
             } else {
                 $cmd = 'cd '.K_LATEX_PATH_PICTURE.' && '.K_LATEX_PDFLATEX.' '.$imgpath.'.tex';
                 $sts = exec($cmd, $out, $ret);
-                if (!$sts) {
+                if (!file_exists($imgpath.'.pdf')) {
                     $error = implode("\n", $out);
                 } else {
                     // convert code using ImageMagick
                     $cmd = 'cd '.K_LATEX_PATH_PICTURE.' && '.K_LATEX_PATH_CONVERT.' -density '.(K_LATEX_FORMULA_DENSITY * $dr).' -trim +repage '.$imgpath.'.pdf -depth 8 -quality 100 '.$imgpath.'.'.K_LATEX_IMG_FORMAT;
                     unset($out);
                     $sts = exec($cmd, $out, $ret);
-                    if ($ret != 0) {
+                    if (!file_exists($imgpath.'.'.K_LATEX_IMG_FORMAT)) {
```

Comment:
Would the following work for you instead?
```suggestion
                   if (($sts === false) || ($ret != 0)) {
```

Reply:
This is working, but I think it's better to also apply this change: [https://github.com/tecnickcom/tcexam/pull/407/files/10546b3e66e38562f4202dfc8c3eb9f8ca76a9fa#r1174262609](url)